e1000e: Replace PREEMPT_RT_FULL with PREEMPT_RT

shruthi-ravi opened the pull request:
Commit a555b1a added a delay after long writes on PREEMPT_RT-enabled
kernels, but using the PREEMPT_RT_FULL option, which has since been
replaced by just PREEMPT_RT.
Therefore, 
 - update the delay condition to check on PREEMPT_RT instead.
 - readd CONFIG_E1000_DELAY to nati_x86_64_defconfig

AzDO WI: [2114543](https://ni.visualstudio.com/DevCentral/_workitems/edit/2114543)

shruthi-ravi committed:
e1000e: Replace PREEMPT_RT_FULL with PREEMPT_RT

Commit a555b1a added a delay after long writes on PREEMPT_RT-enabled
kernels, but using the PREEMPT_RT_FULL option, which has since been
replaced by just PREEMPT_RT.
Therefore, update the delay condition to check on PREEMPT_RT instead.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

gratian commented:
@shruthi-ravi it looks like CONFIG_E1000_DELAY got dropped from `nati_x86_64_defconfig` probably when it was regenerated and because of the PREEMPT_RT_FULL getting renamed. Can you add a commit to re-enable it in defconfig?

Also, can you submit a PR against the 5.15 branch with these changes too since it will be the new default kernel? I will pull this in 5.10 for completion since it affects it.

shruthi-ravi committed:
nati_x86_64_defconfig: readd CONFIG_E1000_DELAY=y

CONFIG_E1000_DELAY was dropped when defconfig was updated from 4.14
in commit 05b9011, because PREEMPT_RT_FULL was replaced by PREEMPT_RT.
Add it back now that the Kconfig depends has been fixed
for this config.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>
